src/clj/puppetlabs/services/jruby/jruby_puppet_schemas.clj

```diff
@@ -77,51 +55,16 @@
    :http-client-cipher-suites [schema/Str]
    :http-client-connect-timeout-milliseconds schema/Int
    :http-client-idle-timeout-milliseconds schema/Int
-   :borrow-timeout schema/Int
-   :max-active-instances schema/Int
-   :max-borrows-per-instance schema/Int
+   :max-requests-per-instance schema/Int
    :use-legacy-auth-conf schema/Bool})
 
-(defn jruby-main-instance?
-  [x]
-  (instance? Main x))
-
-(defn jruby-main-status-instance?
-  [x]
-  (instance? Main$Status x))
-
-(defn jruby-scripting-container?
-  [x]
-  (instance? ScriptingContainer x))
-
-(defn jruby-instance-config?
-  [x]
-  (instance? RubyInstanceConfig x))
-
-(def JRubyMain
-  (schema/pred jruby-main-instance?))
-
-(def JRubyMainStatus
-  (schema/pred jruby-main-status-instance?))
-
-(def ConfigurableJRuby
-  ;; This schema is a bit weird.  We have some common configuration that we need
-  ;; to apply to two different kinds of JRuby objects: `ScriptingContainer` and
-  ;; `JRubyInstanceConfig`.  These classes both have the same signatures for
-  ;; all of the setter methods that we need to call on them (see
-  ;; `jruby-internal/init-jruby-config`), but unfortunately the JRuby API doesn't
-  ;; define an interface for those methods.  So, rather than duplicating the logic
-  ;; in multiple places in the code, we use this (gross) schema to enforce that
-  ;; an object must be an instance of one of those two types.
-  (schema/conditional
-   jruby-scripting-container? (schema/pred jruby-scripting-container?)
-   jruby-instance-config? (schema/pred jruby-instance-config?)))
-
-(def EnvMap
-  "System Environment variables have strings for the keys and values of a map"
-  {schema/Str schema/Str})
-
-(def EnvPersistentMap
-  "Schema for a clojure persistent map for the system environment"
-  (schema/both EnvMap
-    (schema/either PersistentArrayMap PersistentHashMap)))
+(def CombinedJRubyPuppetConfig
```

Comment:
Does this get used anywhere other than from `jruby-puppet-testutils` - in a test namespace?  If not, I'd be inclined not to put this into this namespace, a production code namespace.  Also, it might be helpful to rename this to `JRubyPuppetTkConfig` or something like that to somehow indicate in the name that the config is the TK representation - as opposed a run-time representation that the services use.